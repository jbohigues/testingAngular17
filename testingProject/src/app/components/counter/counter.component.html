<section>
  <h2>Counter {{ counter }}</h2>

  <div>
    <button (click)="increaseBy(-1)">-1</button>
    <button (click)="increaseBy(+1)">+1</button>
  </div>
</section>
