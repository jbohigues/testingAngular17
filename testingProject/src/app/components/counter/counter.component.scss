section,
div {
  display: flex;
  align-items: center;
  gap: 2vw;

  button {
    width: 50px;
    padding: 10px 0;
  }
}
