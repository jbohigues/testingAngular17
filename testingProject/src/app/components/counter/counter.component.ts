import { Component } from '@angular/core';

@Component({
  selector: 'app-counter',
  standalone: true,
  imports: [],
  templateUrl: './counter.component.html',
  styleUrl: './counter.component.scss',
})
export class CounterComponent {
  counter: number = 10;
  increaseBy(value: number) {
    this.counter += value;
    // console.log({ newValue: this.counter });
  }
}
