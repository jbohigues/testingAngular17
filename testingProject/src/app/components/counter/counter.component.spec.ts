import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CounterComponent } from './counter.component';

describe('CounterComponent', () => {
  let component: CounterComponent;
  let fixture: ComponentFixture<CounterComponent>;
  let compiled: HTMLElement;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [CounterComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(CounterComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
    compiled = fixture.nativeElement;
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('match with snapshot', () => {
    expect(compiled).toMatchSnapshot();
  });

  it('increaseBy must increment according to the argument', () => {
    component.increaseBy(5);
    expect(component.counter).toBe(15);
  });

  it('click buttons should increment or decrement in 1', () => {
    const buttons = compiled.querySelectorAll('button');
    buttons[0].click();
    expect(component.counter).toBe(9);
    buttons[1].click();
    buttons[1].click();
    expect(component.counter).toBe(11);
  });

  it('function increaseBy should change text in html', () => {
    component.increaseBy(10);

    // sin esto no funciona, debemos detectar cambios
    fixture.detectChanges();

    const h2 = compiled.querySelector('h2');
    expect(h2?.textContent).toContain('20');
  });
});
