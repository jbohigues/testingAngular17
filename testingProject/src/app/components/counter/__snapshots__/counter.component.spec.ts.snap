// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterComponent match with snapshot 1`] = `
<div
  id="root1"
>
  <section>
    <h2>
      Counter 10
    </h2>
    <div>
      <button>
        -1
      </button>
      <button>
        +1
      </button>
    </div>
  </section>
</div>
`;
